File: Trainer-backend/Trainer-backend/Model/WorkSet.cs
using System.ComponentModel.DataAnnotations;
using System.Runtime.Serialization;
using System.Text.Json.Serialization;

namespace Trainer_backend.Model
{
    [Serializable]
    public class WorkSet: BaseEntity
    {
        [Required]
        [DataMember]
        public string Name { get; set; } = string.Empty;

        [Required]
        [DataMember]
        public virtual List<Set> Sets { get; set; } = new ();

        [DataMember]
        [JsonIgnore]
        public int RoutineId { get; set; }
    }
}

File: Trainer-frontend/src/app/module/workset.ts

export class Workset {
  id: number;
  name: string;
  sets: Sets[];
  routineId: number;

  constructor(id: number = 0, name: string, sets: Sets[], routineId: number) {
    this.id = id;
    this.name = name;
    this.sets = sets;
    this.routineId = routineId;
  }
}
